x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockUtils.java

```diff
@@ -320,7 +320,15 @@ yield new AggregateMetricDoubleLiteral(
             }
             case TDIGEST -> {
                 TDigestBlock tDigestBlock = (TDigestBlock) block;
-                yield tDigestBlock.getTDigestHolder(offset);
+                // return a copy so that the returned value is not bound to the lifetime of the block
+                TDigestHolder blockBacked = tDigestBlock.getTDigestHolder(offset);
+                yield new TDigestHolder(
```

Comment:
Not doing this copy does fun things such as re-using the backing `byte[]` while we still are holding on to it for test-validations.